src/sm/layermanager/layermanager.cpp

```diff
@@ -0,0 +1,459 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/sm/layermanager.hpp"
+#include "aos/common/tools/fs.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace sm {
+namespace layermanager {
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+namespace {
+
+LayerData CreateLayerData(const aos::LayerInfo& layer, const aos::oci::ImageManifest& manifest, const String& path)
+{
+    LayerData layerData = {};
+
+    layerData.mLayerID     = layer.mLayerID;
+    layerData.mLayerDigest = layer.mLayerDigest;
+    layerData.mVersion     = layer.mVersion;
+    layerData.mPath        = path;
+    layerData.mSize        = manifest.mConfig.mSize;
+    layerData.mState       = LayerStateEnum::eActive;
+    layerData.mTimestamp   = Time::Now();
+
+    return layerData;
+}
+
+void AcceptAllocatedSpace(spaceallocator::SpaceItf* space)
+{
+    if (!space) {
+        return;
+    }
+
+    if (auto err = space->Accept(); !err.IsNone()) {
+        LOG_WRN() << "Can't accept memory: err=" << err;
+    }
+}
+
+void ReleaseAllocatedSpace(const String& path, spaceallocator::SpaceItf* space)
+{
+    if (!path.IsEmpty()) {
+        FS::RemoveAll(path);
+    }
+
+    if (!space) {
```

Comment:
ditto, do we need this check at all?

Reply:
yes, null is possible by design to simplify defer release code